### ISSUE TITLE
Is it possible to check for missing type 1 or 2 tags

### ISSUE THREAD

**Issue by hintham** [fo-dicom/fo-dicom#927]
I noticed there is a verify method on DicomItem classes. This method can obviously only run on items present in the dicom object. Does fo-dicom support a way to detect missing type 1 or 2 tags?

**Comment by gofal** [comment]
Currently not. But it would be a great enhancement.

**Comment by hintham** [comment]
What would it take to implement such a feature? I have been searching for a data dictionary to compare an object against but couldn't find anything usable.

**Comment by buckstephenh** [comment]
Did you look at dvtk?  It would be nice to integrate.

**Comment by hintham** [comment]
Good one, the [definition files](https://github.com/dvtk-org/DVTk/tree/master/Definitions/DICOM) seem to contain all required information.

**Comment by rkm** [comment]
I'd also be interested in supporting this. Is there a proposed plan to implement it, or is it free for someone to pick up?

**Comment by gofal** [comment]
I do not have a plan in very near future. So it is free to pick up. Set yourself as assignee and happy coding!

**Comment by mrbean-bremen** [comment]
I don't think this makes sense in fo-dicom. As the type of the tag depends on the SOP class (the same tag may occur with different type in different modules), fo-dicom would need to know the modules used by each SOP class to decide which module the tag belongs to (the modules themselves can also be mandatory, optional, or mandatory under some condition). Additionally, there are the 1C and 2C type tags to consider, for which a specific condition has to checked to know if it is needed or not. I have done something like this as a separate Python project, and I would not consider this trivial.

All that can certainly be done in a separate project, but I suggest to close this issue as it is not suitable for fo-dicom in my opinion.

Ah, and there has been dvtk mentioned (missed this), which is a separate project handling this. Integrating it into fo-dicom is not an option, both because I consider it not practical, and because it is a commercial project (e.g. you need to purchase the data needed for validation).
